src/wwwroot/app/views/candidate/candidate.controller.js

```diff
@@ -1,18 +1,27 @@
-export default function CandidateController($scope, CandidateService) {
+export default function CandidateController($scope, CandidateService, LoggerService) {
    'ngInject';
 
    var urlId = 'Candidates/Put';
    var vm = $scope;
    vm.candidate = {};
-
-   function addCandidate(firstName, lastName, gender) {
-      vm.candidate.firstName = firstName;
-      vm.candidate.lastName = lastName;
-      vm.candidate.gender = gender;
-      var candidateJson = JSON.stringify(vm.candidate);
-      CandidateService.addCandidate(urlId, candidateJson);
+   function _onError(message) {
+      if (message.status === -1) {
```

Comment:
It's bad idea to log errors in controller. Do it in your service.
You need only show some error message in controller